app/models/manageiq/providers/hawkular/middleware_manager/event_catcher/monitor.rb

```diff
@@ -0,0 +1,41 @@
+class ManageIQ::Providers::Hawkular::MiddlewareManager::EventCatcher::Monitor
+  def initialize(ems)
+    @ems               = ems
+    @alerts_client     = ems.alerts_connect
+    @collecting_events = false
+  end
+
+  # Start capturing events
+  def start
+    @collecting_events = true
+  end
+
+  # Stop capturing events
+  def stop
+    @collecting_events = false
+  end
+
+  def each_batch
+    while @collecting_events
+      yield fetch
+    end
+  end
+
+  private
+
+  # Each fetch is performed from the time of the most recently caught event or 1 minute back for the first poll.
+  # This gives us some slack if hawkular events are timestamped behind the miq server time.
+  # TODO: Is there a danger here?  This assumes all Hawkular events at time T are fetched in one call.
```

Comment:
I didn't investigate or try to reproduce.  I'm not sure it's a huge concern at the moment, Hawkular is not generating massive numbers of events and as such it is unlikely that there would be more than one for the same millisecond, and that the query would be performed in the midst of writes for the same ms. It may be a feasible scenario but I think it's unnecessary to try and handle it at this time.  Other providers would have this same issue, I think.